src/CSharpExtDownloader.ts

```diff
@@ -46,95 +39,52 @@ export class CSharpExtDownloader
 
         let telemetryProps: any = {};
 
-        return util.touchInstallFile(util.InstallFileType.Begin)
-            .then(() => {
-                installationStage = 'getPlatformInfo';
-                return PlatformInformation.GetCurrent();
-            })
-            .then(info => {
-                platformInfo = info;
-                packageManager = new PackageManager(info, this.packageJSON);
-                this.logger.appendLine();
-
-                // Display platform information and RID followed by a blank line
-                this.logger.appendLine(`Platform: ${info.toString()}`);
-                this.logger.appendLine();
-
-                installationStage = 'downloadPackages';
-
-                const config = vscode.workspace.getConfiguration();
-                const proxy = config.get<string>('http.proxy');
-                const strictSSL = config.get('http.proxyStrictSSL', true);
-
-                return packageManager.DownloadPackages(this.logger, status, proxy, strictSSL);
-            })
-            .then(() => {
-                this.logger.appendLine();
+        try {
+            await util.touchInstallFile(util.InstallFileType.Begin);
+            installationStage = 'getPlatformInfo';
+            platformInfo = await PlatformInformation.GetCurrent();
 
-                installationStage = 'installPackages';
-                return packageManager.InstallPackages(this.logger, status);
-            })
-            .then(() => {
-                installationStage = 'touchLockFile';
-                return util.touchInstallFile(util.InstallFileType.Lock);
-            })
-            .then(() => {
-                installationStage = 'completeSuccess';
-                success = true;
-            })
-            .catch(error => {
-                if (error instanceof PackageError) {
-                    // we can log the message in a PackageError to telemetry as we do not put PII in PackageError messages
-                    telemetryProps['error.message'] = error.message;
+            packageManager = new PackageManager(platformInfo, this.packageJSON);
 
-                    if (error.innerError) {
-                        errorMessage = error.innerError.toString();
-                    } else {
-                        errorMessage = error.message;
-                    }
+            // Display platform information and RID followed by a blank line
+            this.logger.appendLine(`Platform: ${platformInfo.toString()}`);
+            this.logger.appendLine();
 
-                    if (error.pkg) {
-                        telemetryProps['error.packageUrl'] = error.pkg.url;
-                    }
+            installationStage = 'downloadPackages';
 
-                } else {
-                    // do not log raw errorMessage in telemetry as it is likely to contain PII.
-                    errorMessage = error.toString();
-                }
+            let networkObject = GetNetworkDependencies();
```

Comment:
how about:
`let networkConfiguration = GetNetworkConfiguration();`? Rename the variable and the method? this seems clearer.